### DISCUSSION TITLE
Conflict with phpinsights: or I get the phpstan error, or I get a small code quality score on Insights

### DISCUSSION THREAD

**Discussion by giacomomasseron** [larastan/larastan#1984] [Q&A]
Hi,  
I use both insights and larastan together, and they go in kinda conflict.
I have this function that returns a specific Model:

```php
    public function handle(): User
    {
        return User::create([...]);
    }
```

This code returns a larastan error:

```
Method App\UseCases\TestUseCase::handle() should return App\Models\Users\User
 but returns Illuminate\Database\Eloquent\Model.
```

If I change the code like this:

```php
    public function handle(): User
    {
        /** @var User $user */
        $user = User::create([]);
        return $user;
    }
```

I do not get larastan error, but insights suggests me to comeback to previous code.

Any thoughts about it? How can I solve it? Thanks.

**Comment by szepeviktor** [comment]
Could share `TestUseCase::handle`?

**Reply by giacomomasseron** [reply]
Sure, I wrote bad, sorry.

```php
    public function handle(): User
    {
		// solution 1
        /** @var User $user */
        $user = User::create([...]);
		return $user;

		// solution 2
        return User::create([...]);
    }
```

**Reply by szepeviktor** [reply]
Where is `new User(...)`?

**Reply by giacomomasseron** [reply]
I updated the initial question. I hope now it's more clear. Thanks for your reply.

**Comment by szepeviktor** [comment]
We have passing tests.
https://github.com/larastan/larastan/blob/91c3e56a9d08418e7bd4cb2284563cb69c0f8212/tests/Rules/data/CorrectModelInstantiation.php#L12-L20

It must be something in your setup.

**Reply by calebdw** [reply]
Are you actually using Larastan or do you just have it installed? What's your phpstan.neon conf?

**Reply by giacomomasseron** [reply]
I am actually using larastan, with the standard phpstan.neon conf file:

	includes:
		- vendor/larastan/larastan/extension.neon

	parameters:

		paths:
			- app/

		# Level 9 is the highest level
		level: 5

	#    ignoreErrors:
	#        - '#PHPDoc tag @var#'
	#
	#    excludePaths:
	#        - ./*/*/FileToBeExcluded.php
	#
	#    checkMissingIterableValueType: false

What I saw that my User class does not extends directly laravel Model, but is a grandchild of Model class.

**Reply by szepeviktor** [reply]
> my User class

Let's see that!

**Reply by giacomomasseron** [reply]
We have long structure, I try to summarize:

    class User extends BaseUser implements CreatorInterface, DisableableInterface, HasCreatorInterface, SoftDeletesInterface

This is BaseUser class:

    abstract class BaseUser extends BaseModel implements AuthenticatableContract, AuthorizableContract, CanResetPasswordContract

This is BaseModel class:

    abstract class BaseModel extends MyCmpModel implements HasActivityLogInterface, HasDtoInterface, IndexableInterface, ShowableInterface

This is MyCmpModel class:

    use Illuminate\Database\Eloquent\Model;

    abstract class CruratedModel extends Model implements ModelInterface

So in the end `User` should extends laravel `Model` class.

**Reply by calebdw** [reply]
That's good, I asked because I just came across an instance where someone had installed larastan with composer but never imported the config into phpstan (or used the extension installer) so it wasn't taking effect

**Reply by calebdw** [reply]
@giacomomasseron, do you override the create method anywhere?

**Reply by giacomomasseron** [reply]
No, I do not override.

**Reply by calebdw** [reply]
There's not much for us to go on here since our tests are passing, can you stand up a new laravel project that contains the minimum reproducible example?
